src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -219,6 +219,14 @@ private void MethodChecks(MethodDeclarationSyntax syntax, Binder withTypeParamsB
                     var loc = parameterSyntax.Type.Location;
                     diagnostics.Add(ErrorCode.ERR_BadTypeforThis, loc, parameter0Type);
                 }
+                else if (this.Parameters[0].RefKind == RefKind.Ref && !parameter0Type.IsValueType)
+                {
+                    diagnostics.Add(ErrorCode.ERR_RefExtensionMustBeValueTypeOrConstrainedToOne, location, Name);
+                }
+                else if (this.Parameters[0].RefKind == RefKind.RefReadOnly && parameter0Type.TypeKind != TypeKind.Struct)
```

Comment:
Yes. we have such methods in `RefExtensionMethods.cs`

---
In reply to: [120740369](https://github.com/dotnet/roslyn/pull/19703#discussion_r120740369) [](ancestors = 120740369)